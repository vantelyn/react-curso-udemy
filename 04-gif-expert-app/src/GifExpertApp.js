import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';


const GifExpertApp = ( { defaultCats = [] } )=>{

    // const [categories, setCategories] = useState(['One Punch']);
    const [categories, setCategories] = useState( defaultCats );

    const handleReset = () => setCategories([]);

    return(
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories } />
            <button onClick={ handleReset }>Reset</button>
            <hr />

            <ol>
                {
                    categories.map( category => 
                        <GifGrid
                            key={ category }
                            category={ category } 
                        />
                    )
                }
            </ol>

        </>
    );

}

export default GifExpertApp;