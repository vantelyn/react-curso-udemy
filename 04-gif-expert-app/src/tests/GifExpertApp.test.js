import GifExpertApp from "../GifExpertApp";
import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';


describe('Pruebas en <GifExpertApp />', () => { 

    test('debe renderizar el componente adecuadamente', () => { 
        const wrapper = shallow(<GifExpertApp />);
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de mostrar una lista de categorÃ­as', () => { 
        const categories = [ 'One Punch','Gay' ];
        const wrapper = shallow(<GifExpertApp  defaultCats={ categories } />);
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find( 'GifGrid' ).length ).toBe( categories.length );
    })

});