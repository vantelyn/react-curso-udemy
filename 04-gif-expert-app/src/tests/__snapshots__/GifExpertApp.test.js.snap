// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> debe de mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Gay"
      key="Gay"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> debe renderizar el componente adecuadamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <hr />
  <ol />
</Fragment>
`;
